src/runtime/components/NaiveMenuLink.vue

```diff
@@ -18,11 +18,18 @@ interface NaiveMenuLinkProps
 }
 
 const props = defineProps<NaiveMenuLinkProps>();
-const activeKey = ref(getActiveKey(useRoute().path));
 
-useNuxtApp().$router.afterEach((to) => {
-    activeKey.value = getActiveKey(to.path)
-});
+const router = useRouter()
+
+// check if deprecated path logic is used
+const isDeprecatedKey = computed(() => props.routes.some(checkPath))
+const checkPath = (m: MenuLinkRoute): boolean => !!m.path || m.children?.some(checkPath) || false
+
+// The menu's active key
+const activeKey = computed(() => isDeprecatedKey.value
+    ? getActiveKey(router.currentRoute.value?.path)
+    : router.currentRoute.value?.name?.toString()
```

Comment:
I'm sorry for taking too long to submit the review 🙏. I thought I did but it was pending.

As the proposed approach relies on route name to compute the menu's active key, it would break the current logic. For example, two routes `/products` and `/products/:id` have names `products` and `products-id` thus cannot both be active at the same time which is not the case with the existant approach.

Reply:
To be honest, that shouldn't really be a problem. The active key of the menu can only ever be one anyway. The fact that `product` and `product-id` cannot both be active should not really be the case. If `product-id` is a child of `product` and `product-id` is active, `product` is also displayed as active. (see: https://www.naiveui.com/en-US/os-theme/components/menu#expand-selected-option.vue)

What could actually be a problem are route params/querys.

Let's say you have the following menu structure:

```
- product
    | - product/1
    | - product/2
```
it would not be possible to distinguish clearly by name. The question is, of course, whether this is a realistic scenario for linking specific params in the menu.

The problem with the current logic is that if I have a link on `/` and another on `/page`, the `/` route is always shown as active during a refresh.

Reply:
The `NaiveMenuLink` component is intended to be used internally by layout components. That's why is not mentioned on docs. These navigation components (navbar/sidebar) expect known static routes. That's why the active key is computed from route's path and not name. In case the path is not defined on the `routes` prop for example `/products/{id}` it will [fallback](https://github.com/becem-gharbi/nuxt-naiveui/blob/4dd39a9a3aafa44c04d623712bb70d309be6842f/src/runtime/components/NaiveMenuLink.vue#L36) to the parent route for example `/products`.

> The problem with the current logic is that if I have a link on / and another on /page, the / route is always shown as active during a refresh.

Can you provide more details about this issue or a reproduction maybe?

Reply:
We are talking about two use cases of `NaiveMenuLink`. The one I see is for primary navigation such as a Navbar. This means that the links are static and there should always be an active link (directly or indirectly). The one you see is for secondary navigation. This means that the links can be dynamic thus the named routing with params.

The feature added here is to support the second use case without deprecating the first. I suggest adding a new prop, eg `active-by`, which can be `path` for the first use case and `name` for the second. 

What do you think?

Reply:
Ill have a look into it, tomorrow